chainerx_cc/chainerx/routines/indexing.cc

```diff
@@ -198,4 +198,41 @@ Array Take(const Array& a, const Array& indices, int8_t axis) {
     return out;
 }
 
+Array Where(const Array& condition, const Array& x, const Array& y) {
+    const Array& condition_cast = condition.dtype() != Dtype::kBool ? condition.AsType(Dtype::kBool) : condition;
```

Comment:
Could you move this cast to kernel definition of each device?

Reply:
Sure.